units/patch_intent.go

```diff
@@ -308,7 +308,7 @@ func (j *patchIntentProcessor) finishPatch(ctx context.Context, patchDoc *patch.
 		var patchConfig *model.PatchConfig
 		repeatPatchID, shouldRepeat := j.intent.RepeatPreviousPatchDefinition()
 		if shouldRepeat {
-			patchDoc.ReferencePatchID = repeatPatchID
+			patchDoc.ReferenceManifestID = repeatPatchID
```

Comment:
Will a patch always have a manifest associated with it, even if it has no modules? If so, the logic for using the reference manifest ID to get the modules could error out because there's no modules to be loaded. It might also be a bit weird to set a manifest ID even if no such manifest exists, so it would be worth checking explicitly that the manifest doc really does exist before setting the reference manifest ID.

Reply:
Just the comment saying it might not exist is probably good enough. The usage of the field is already pretty limited anyways.